sqlmesh/core/model/decorator.py

```diff
@@ -18,16 +24,26 @@
 logger = logging.getLogger(__name__)
 
 
+DECORATOR_RETURN_TYPE = t.TypeVar("DECORATOR_RETURN_TYPE")
+
+
 class model(registry_decorator):
     """Specifies a function is a python based model."""
 
     registry_name = "python_models"
     _dialect: DialectType = None
 
-    def __init__(self, name: str, is_sql: bool = False, **kwargs: t.Any) -> None:
-        if not name:
+    def __call__(
+        self, func: t.Callable[..., DECORATOR_RETURN_TYPE]
+    ) -> t.Callable[..., DECORATOR_RETURN_TYPE]:
+        self.name = self.name or get_model_name(Path(inspect.getfile(func)))
+
+        if not self.name:
             raise ConfigError("Python model must have a name.")
```

Comment:
1. Can this ever _not_ be set after these changes?
2. The `ConfigError` doesn't really make sense anymore, since we supposedly don't require the model to have a name anymore.

Reply:
If we do it like this, we should move the `kind is not None` branch within `__call__`, otherwise `name` would be interpolated as `None` in the warning.

Reply:
Yes correct I'm moving this part of code to `__call__`. The name should also be derived always from the filename so the check is redundant.